includes/payment-methods/class-wc-stripe-upe-payment-method-affirm.php

```diff
@@ -50,4 +50,15 @@ public function __construct() {
 	public function requires_automatic_capture() {
 		return false;
 	}
+
+	/**
+	 * Returns whether the payment method is available for the Stripe account's country.
+	 *
+	 * Affirm is only available domestic transactions in the United States or Canada.
+	 *
+	 * @return bool True if the payment method is available for the account's country, false otherwise.
+	 */
+	public function is_available_for_account_country() {
+		return in_array( WC_Stripe::get_instance()->account->get_account_country(), $this->supported_countries, true );
```

Comment:
Would it be worth renaming the `$this->supported_countries` property, and introducing a new one for supported account countries?

The property name seems ambiguous because it could refer to either supported billing countries or Stripe account countries.

Before BNPL, it was used to check for supported billing countries. 
Ref [#](https://github.com/woocommerce/woocommerce-gateway-stripe/blame/develop/includes/payment-methods/class-wc-stripe-upe-payment-method.php#L280-L294), [#](https://github.com/woocommerce/woocommerce-gateway-stripe/blame/develop/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L671), [#](https://github.com/woocommerce/woocommerce-gateway-stripe/blame/develop/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L2245), [#](https://github.com/woocommerce/woocommerce-gateway-stripe/blame/develop/includes/abstracts/abstract-wc-stripe-payment-gateway-voucher.php#L258)

And for BNPLs, we're also using it to check supported Stripe account countries.